packages/eslint-plugin/tests/rules/prefer-nullish-coalescing.test.ts

```diff
@@ -751,5 +759,27 @@ declare const c: ${type};
         },
       ],
     })),
+    ...ignorablePrimitiveTypes.map<
+      TSESLint.InvalidTestCase<MessageIds, Options>
+    >(ignoreablePrimitive => ({
+      code: `
+declare const x: ${ignoreablePrimitive} | undefined;
+x || y;
+      `,
+      options: [
+        {
+          ignorePrimitives: Object.fromEntries(
+            ignorablePrimitiveTypes
+              .filter(t => t !== ignoreablePrimitive)
+              .map(t => [t, true]),
+          ),
+        },
+      ],
+      errors: [
+        {
+          messageId: 'preferNullishOverOr',
+        },
+      ],
+    })),
```

Comment:
Some missing test cases to fill in:

* Literal types (`true`, `1`, etc.)
* Template literals (<code>\`hello${string}\`</code>)
* Bigints (`1n`)
* Combinations of the types (`boolean | string | ...`), etc.